conda_anaconda_tos/api.py

```diff
@@ -325,3 +334,251 @@ def clean_tos(tos_root: str | os.PathLike[str] | Path) -> Iterator[Path]:
+        backup_configs_dir = timestamped_backup_dir / "configs"
+        try:
+            backup_configs_dir.mkdir(exist_ok=True)
+        except (PermissionError, OSError) as e:
+            raise CondaToSBackupError(
+                "create configs directory", backup_configs_dir, str(e)
+            ) from e
+
+        for info in file_info:
+            source_location = Path(info.source_location).expanduser().resolve()
+
+            try:
+                tos_file_resolved = info.file_path.resolve()
+                # Create a path structure that preserves source location info
+                rel_path = tos_file_resolved.relative_to(source_location)
+                # Include source identifier in backup path
+                source_hash = get_location_hash(source_location)
+                backup_file = backup_configs_dir / source_hash / rel_path
+                backup_file.parent.mkdir(parents=True, exist_ok=True)
+                shutil.copy2(info.file_path, backup_file)
+            except (ValueError, PermissionError, OSError):
+                # File issues, skip but continue with other files
+                continue
+
+    # Create metadata file
+    metadata = {
+        "timestamp": backup_timestamp.isoformat(),
+        "backup_type": "comprehensive",
+        "files_count": len(file_info),
+        "source_locations": list({info.source_location for info in file_info}),
+        "temporary_locations": list(
+            {info.source_location for info in file_info if info.is_temporary}
+        ),
+        "persistent_locations": list(
+            {info.source_location for info in file_info if not info.is_temporary}
+        ),
+        "files": [
+            {
+                "path": str(info.file_path),
+                "source_location": info.source_location,
+                "is_temporary": info.is_temporary,
+            }
+            for info in file_info
+        ],
+        "version": APP_VERSION,
+    }
+
+    metadata_file = timestamped_backup_dir / "metadata.json"
+    try:
+        with metadata_file.open("w") as f:
+            json.dump(metadata, f, indent=2)
+    except (PermissionError, OSError) as e:
+        raise CondaToSBackupError("write metadata", metadata_file, str(e)) from e
+
+    return timestamped_backup_dir
+
+
+def restore_tos_configs(
+    timeout_hours: float = 1.0,
+) -> tuple[bool, list[Path]]:
+    """Restore ToS configuration files from recent backup to exact original locations.
+
+    Args:
+        timeout_hours: Timeout in hours for considering backups recent
+
+    Returns:
+        - bool: True if any restore was performed, False otherwise
+        - list[Path]: List of successfully restored files
+
+    Raises:
+        FileNotFoundError: If the backup directory does not exist
+
+    """
+    backup_dir = TOS_BACKUP_DIR
+
+    if not backup_dir.exists():
+        raise FileNotFoundError(f"Backup directory does not exist: {backup_dir}")
```

Comment:
I think this will cause an installation failure since `conda tos restore` is run as a post-link script. When installed for the first time, that backup directory does not exist.